### PR TITLE
Test cypress travis

### PR DESCRIPTION
**Pull request by agduncan94**
* Cypress is now used for testing instead of Protractor
* Coverage reporting is currently removed until we find a solution for Cypress
- [x] Check that you pass the basic compilation and unit tests by running `grunt`

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -16,21 +21,39 @@ before_install:
...
   - 'sh -e /etc/init.d/xvfb start'
+  - 'sudo add-apt-repository -y ppa:webupd8team/java'
+  - 'sudo apt-get update'
+  - 'sudo apt-get install -y libappindicator1 fonts-liberation'
+  - 'sudo apt-get install -y oracle-java8-installer'
+  - 'sudo apt install -y oracle-java8-set-default'
+  - 'wget https://dl.google.com/linux/direct/google-chrome-stable_current_amd64.deb'
+  - 'sudo dpkg -i google-chrome*.deb'
+  - 'npm install -g cypress-cli' #test using cypress cli for integrationg testing
+  - 'wget https://seqwaremaven.oicr.on.ca/artifactory/collab-release/io/dockstore/dockstore-webservice/1.1-rc.2/dockstore-webservice-1.1-rc.2.jar'
```
Comment (denis-yuen) [inline .travis.yml]: Change this to 1.1 and make it an environment variable so it is easy to change https://docs.travis-ci.com/user/environment-variables/

File: .travis.yml [hunk 2]
```diff
@@ -16,21 +21,39 @@ before_install:
...
-after_script:
-- cat ./build/coverage/**/lcov.info | ./node_modules/coveralls/bin/coveralls.js
-after_success: grunt coveralls
+
+before_script:
+  - grunt # this does syntax checking, unit testing, and jshint
+  - grunt serve --silent & #make sure it runs the server first before testing
+  - 'sleep 10'
+  - psql 'postgres' -U postgres < travisci/db_dump.sql
+  - java -jar dockstore-webservice-*.jar server travisci/web.yml 1>/dev/null &
```
Comment (denis-yuen) [inline .travis.yml]: Also use the environment variable here

File: .travis.yml [hunk 3]
```diff
@@ -16,21 +21,39 @@ before_install:
...
-  - 'node_modules/protractor/bin/webdriver-manager start --standalone --firefox &'
-  - 'sleep 5'
-after_script:
-- cat ./build/coverage/**/lcov.info | ./node_modules/coveralls/bin/coveralls.js
-after_success: grunt coveralls
+
+before_script:
+  - grunt # this does syntax checking, unit testing, and jshint
+  - grunt serve --silent & #make sure it runs the server first before testing
+  - 'sleep 10'
```
Comment (denis-yuen) [inline .travis.yml]: Make a ticket to use dockerize or similar here, this looks fragile depending on performance

File: cypress/integration/workflowDetails.js [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+        .get("#dag_fullscreen")
+          .click()
+          .get("#dag-col")
+            .should("have.class", "fullscreen-element")
+            .get("#dag_fullscreen")
+              .click()
+              .should("not.have.class", "fullscreen-element")
+    });
+  });
+})
```
Comment (denis-yuen) [inline cypress/integration/workflowDetails.js]: What is the image below for?

File: cypress/support/commands.js [hunk 5]
```diff
@@ -0,0 +1,39 @@
+// ***********************************************
```
Comment (denis-yuen) [inline cypress/support/commands.js]: Why is there a totally commented file here?

File: cypress/support/defaults.js [hunk 6]
```diff
@@ -0,0 +1,17 @@
+// ***********************************************
```
Comment (denis-yuen) [inline cypress/support/defaults.js]: same

File: cypress/integration/workflowDetails.js (hunk 4 above)
Comment (agduncan94) [inline cypress/integration/workflowDetails.js]: It seems that we have more of the cypress files committed than necessary. I'll add the appropriate folders to the git ignore

File: .travis.yml [hunk 7]
```diff
@@ -6,6 +6,8 @@ node_js:
   - '4.2.1'
 services:
   - postgresql
+jdk:
+  - oraclejdk8
 env:
   - WEBSERVICE_VERSION="1.1-rc.2"
```
Comment (denis-yuen) [inline .travis.yml]: Update to 1.1 rather than 1.1-rc.2

**Comment by denis-yuen** [comment]
This is close, let's get together a how-to for running this from the POV of a new developer with a clean Ubuntu machine. Re-use parts of Dockstore Primer as required. Thanks!

**Comment by agduncan94** [comment]
I've been updating the Dockstore Primer to include a how to for a new developer, but still have more to add.

**Comment by denis-yuen** [comment]
Ok, got it to work actually. 
I think that you should document the dropwizard hibernate setting and the cypress.json supportFile issue, but otherwise it looks good.

**Comment by agduncan94** [comment]
I'll go through the rest of the options for configuration to ensure that there aren't any other weird defaults, and then add to the primer information about the configuration file.

**Comment by denis-yuen** [comment]
Thanks @jpatricia  (and @agduncan94 ) by the way

**Pull request merged**